### ISSUE TITLE
Adapting to changed font size

### ISSUE THREAD

**Issue by jdujava** [jstkdng/ueberzugpp#220]
As described in https://github.com/gokcehan/lf/issues/1793, I use `ueberzugpp` for image previews in `lf`. After changing the font size, `lf` outputs updated position and size of preview (measured in rows/columns), but the preview is displayed as if the font size was the original one, resulting in wrong position/sizing.

Is there a way to update `ueberzugpp` about the change in font size?

**Comment by jstkdng** [comment]
hey, i don't think there's a way for `ueberzugpp` to react to that event, there could be one I'd have to dig through some documentation.

You could trigger a redraw on font size change maybe? Does lf support that?

**Comment by jdujava** [comment]
I think `lf` triggers a redraw on font size change. Maybe I should specify earlier: all subsequent image previews are positioned/sized wrong.

**Comment by joelim-work** [comment]
I spent some time to investigate, `lf` does detect terminal resizes and in response sends an update to `ueberzugpp` with the adjusted cell dimensions. However it appears that `ueberzugpp` calculates the pixel dimensions only once and then applies it for all subsequent updates, even if the terminal has resized.

Below are the steps to reproduce without involving `lf`:

1. Start `ueberzugpp` (should read JSON commands from `stdin`):
   ```sh
   ueberzug layer
   ```
2. Type the following to display the image:
   ```json
   {"action": "add", "identifier": "test", "path": "foobar.jpg", "width": 10, "height": 10, "x": 2, "y": 2}
   ```
3. Resize the terminal. At this point, it is expected that the image does not change, because `ueberzugpp` hasn't received any updates.
4. Repeat step 2. Nothing happens, but now that another `add` command has been sent to `ueberzugpp`, I would expect the image to be resized and repositioned according the the new cell size.

I also tested this on (a continuation of) the original [`ueberzug`](https://github.com/ueber-devel/ueberzug), and the image updated fine even after resizing the terminal.

**Comment by jstkdng** [comment]
It does? I think it recalculates terminal sizes on every command (calling ioctl) whilst `ueberzugpp` just uses the first value it calculates. Right now there's no trivial way to do that I believe.

**Comment by joelim-work** [comment]
That's my assumption based on what I found when I was trying to debug the issue myself. So just to confirm, are you or @jdujava able to reproduce the behavior I described above for both `ueberzugpp` and `ueberzug`?

**Comment by jdujava** [comment]
Yeah, I tried following your steps, and `ueberzug` properly updated both position and size of the image, while `ueberzugpp` didn't.

_This is maybe more suited to mention in `lf` issue, but position/size of previews in `lf` didn't update with neither `ueberzug` or `ueberzugpp`. But this can be caused by my setup and way of launching `lf` with `ueberzug`, more than with their response to resizing._

**Comment by jdujava** [comment]
Just to clear up my previous message:
As I described in https://github.com/gokcehan/lf/issues/1793, with proper setup `ueberzug` properly adapts to terminal font resize.
On the other hand, as we already know, `ueberzugpp` is stuck with the initial font size.

**Comment by jstkdng** [comment]
eyyyy @seebye, nice seeing you, thanks for the tip

@jdujava this issue has been solved on this repo: https://github.com/jstkdng/ueberzugpp-new, I'll see if I can implement SIGWINCH
